proxy/proxy.go

```diff
@@ -1132,6 +1132,8 @@ func (p *Proxy) do(ctx *context) (err error) {
 		backendStart := time.Now()
 		rsp, perr := p.makeBackendRequest(ctx, backendContext)
 		if perr != nil {
+			p.handleFiltersOnError(ctx, processedFilters)
```

Comment:
Would it be possible to create a response similar to what `p.errorResponse` does and set it as a response in the context?

Reply:
Something like
```go
rsp, perr := p.makeBackendRequest(ctx, backendContext)
if perr != nil {
    ctx.response = makeErrorResponse(perr)
    p.handleFiltersOnError(ctx, processedFilters)
    ...
```
where `makeErrorResponse` replicates data/logic from `p.errorResponse`+`p.sendError` and `p.sendError` then would just send `ctx.response`.

Reply:
@AlexanderYastrebov that looks like it can't work. `p.errorResponse` and `p.sendError` require access to ctx and `p`. I think then we would need to do:
```
if perr != nil {
    p.makeErrorResponse(ctx, perr)
    p.handleFiltersOnError(ctx, processedFilters)
```

Reply:
Do you mean that all changes/writes to ctx.responseWriter in `p.errorResponse`+`p.sendError` shouldbe replicated to create a response?
Looking into the methods show up more questions than answers to your comment, so please clarify. :)